### ISSUE TITLE
Some another ideas, if you will allow.

### ISSUE THREAD

**Issue by CenturyCenturion** [nipkownix/re4_tweaks#551]
### Describe the feature you'd like to see added

Hello Nipkownix! If you allow: some my ideas for Re4_tweaks again. And again English not my Native. Sorry.

1. Please fix mouse aiming with harpoon during Del Lago boss battle (make aiming more accurate, smoother and slightly slower, please).

2. Please add function for rebind key bindings on map (zoom etc).

3. Please add an optional fix for "Yes/No" or "Yes/No/Use" mouse selector when taking an item and save/loading menu, like in your Fix "Retry/Load" mouse selector.

4. Maybe it's not important, but it seems to me that the keys, treasures, plants (and maybe other items) when taking (rotating view) looks worse than the same items in the "Examine" menu. Bug or strange feature?

5. Maybe it's not important, but please add an optional fix for the ammo bug (refilling all ammo in weapon's clip after upgrade).

6. Please fix Matilda image in attache case and Merchant's shop interface (this icon has low res and bad quality in comparison with other weapon icons).

7. And another very optional idea about Matilda. But maybe...

This pistol is a bonus weapon and we can take it only after first playthrough. But unlike other bonus weapon (like Chicago Typewriter) Matilda not overpowered weapon. And I have long had a desire to go through the game using Matilda like main pistol.

But, if I use "Cleared Game/NEW ROUND" appearing after first playthrough and buy Matilda from Merchant, I'll get already upgraded Leon's HP, already upgraded attache case and immediately open path for upgrading Matilda up to the maximum. It's not so interesting to play like that.

And I used your Re4_tweaks Trainer section for get the Matilda instead of Handgun in regular playthrough (without "Cleared Game/NEW ROUND"). But in this case Merchant can't upgrade the Matilda.

My request. If you find it possible, please, add an option for upgrading Matilda step by step (similarly other normal weapon) in regular "first" playthrough. I think that the time of upgrades access for Matilda can be done at the same time the Red9 pistol upgrade access (buying and first upgrade - Merchant in underground passage after obtaining Ashley in the Church, final "special" upgrade - Merchant after killing Ramon Salazar).

Thank you for reading this sheet of my (ridiculous) thoughts. Hope this ideas will be useful. And thank you again!

**Comment by nr1995** [comment]
I don't think no. 5 is a bug.
I think it's intended that way so you can immediately see the difference your upgrade made.

**Comment by juuuulian-s** [comment]
No. 5 is definitely a feature, dating as far back as the original RE2. I wouldn't be against an option to disable it tho, for added challenge.
Regarding no. 6, an option to see Matilda, Chicago Typewriter, and Handcannon upgrades gated by campaign progression like the base weapons would also be nice to see.
No. 4 is simply an issue with how the vanilla game handles world models. This is fixed across the board by the HD Project, which also takes care of no. 6
